src/main/scala/cromwell/engine/CallActor.scala

```diff
@@ -2,88 +2,87 @@ package cromwell.engine
-        val inputs = symbolStore.locallyQualifiedInputs(call)
-        val tryCommandLine = call.task.command.instantiate(inputs)
-        tryCommandLine match {
-          case Success(commandLine) =>
-            _creator = Option(sender())
-            self ! Run(commandLine, backend, call, symbolStore)
-            creator ! Started(call)
-            context.become(started)
-          case failure =>
-            sender ! failure
-        }
-
-      case bad@_ if !isStarted =>
-        sender ! InvalidOperation(s"Received message $bad, but not yet Started.")
-    }
+  override def receive = LoggingReceive {
+    case CallActor.Start => handleStart()
+    case badMessage =>
+      val diagnostic = s"Received unexpected message $badMessage."
+      log.error(diagnostic)
+      context.parent ! CallActor.Failed(call, diagnostic)
   }
 
-  private def started: Receive = {
-    LoggingReceive {
-      case Run(commandLine, backend, call, symbolStore) =>
-        log.info(self + " executing command: " + commandLine)
-        val tryOutputs = backend.executeCommand(commandLine, call, call.task.outputs, symbolStore)
-        val (successes, failures) = tryOutputs.partition { _._2.isSuccess }
-
-        if (failures.isEmpty) {
-          // Materialize the Successes.
-          val outputs = successes.map { case (key, value) => key -> value.get }
-          creator ! Done(call, outputs)
-        } else {
-          val errorMessages = TryUtil.stringifyFailures(failures.values).mkString("\n")
-
-          log.error(errorMessages)
-          creator ! Failed(call, errorMessages)
-        }
-
-      case unknown @_ =>
-        sender ! InvalidOperation(s"Unexpected message $unknown.")
+  /**
+   * Performs the following steps:
+   * <ol>
+   *   <li>Collects the locally qualified inputs from the store actor.</li>
+   *   <li>Instantiates the command line using these inputs.</li>
+   *   <li>If the above completes successfully, messages the sender with `Started`,
+   *   otherwise messages `Failed`.</li>
+   *   <li>Executes the command with these inputs.</li>
+   *   <li>Collects outputs in a `Map[String, Try[WdlValue]]`.</li>
+   *   <li>If there are no `Failure`s among the outputs, messages the parent actor
+   *   with `Completed`, otherwise messages `Failed`.</li>
+
+   * </ol>
+   */
+  private def handleStart(): Unit = {
+    // Record the original sender here and not in the yield over the `Future`, as the
+    // value of sender() when that code executes may be different than what it is here.
+    val originalSender = sender()
+
+    for {
+      inputs <- (storeActor ? StoreActor.GetLocallyQualifiedInputs(call)).mapTo[Map[String, WdlValue]]
+      commandLine <- Future.fromTry(call.task.command.instantiate(inputs))
+    } yield {
+      originalSender ! CallActor.Started(call)
+      val tryOutputs = backend.executeCommand(commandLine, call, call.task.outputs, s => inputs.get(s).get)
+      val (successes, failures) = tryOutputs.partition {
```

Comment:
Am I correct in remembering that we have this pattern in multiple places?

It might be nice to have a func which takes a collection of `Try` and two closures (e.g. `Seq[Try] => Unit`) and does basically this.